xcode/Subconscious/Shared/Services/DataService.swift

```diff
@@ -481,6 +481,25 @@ actor DataService {
         _ text: String,
         audience: Audience
     ) async -> MemoAddress? {
+        let prefix = text.prefix(512)
+        let subtext = Subtext(markup: prefix.toString())
+        
+        // Find the first text block
+        let block = subtext.blocks.first { b in
+            switch (b) {
+            case .text:
+                return true
+            case _:
+                return false
+            }
+        }
+        
+        guard let block = block else {
+            return nil
+        }
+        
+        let text = block.body().toString()
```

Comment:
This does not address the capitalisation of the string, I'm not sure where we landed on that? If the first line starts with a capital this will generate a slug with a capital letter but it seems like leaving and reloading the note causes it to become lowercase?